org.lflang/src/org/lflang/ASTUtils.java

```diff
@@ -611,6 +542,29 @@ public static String getUniqueIdentifier(Reactor reactor, String name) {
         return superClasses(reactor, new LinkedHashSet<Reactor>());
     }
 
+    /**
+     * Collect elements of type T from the class hierarchy defined by
+     * a given reactor definition.
+     * @param definition The reactor definition.
+     * @param elements A function that maps a reactor definition to a list of
+     *                 elements of type T.
+     * @param <T> The type of elements to collect (e.g., Port, Timer, etc.)
+     * @return
+     */
+    public static <T> List<T> collectElements(Reactor definition, Function<Reactor,List<T>> elements) {
+        List<T> result = new ArrayList<T>();
+        // Add elements of the current reactor.
```

Comment:
This change looks great! The only "mistake" that I could spot is that the elements of the current reactor are added before for the elements of the super classes. In the original code, it was added after the super class elements. I don't know whether the order within the list has any meaning in this context. If it should not have meaning, then maybe the bug lies somewhere else.